src/java/org/apache/cassandra/auth/CassandraRoleManager.java

```diff
@@ -135,6 +135,12 @@ static int getGensaltLogRounds()
     // Will be set to true when all nodes in the cluster are on a version which supports roles (i.e. 2.2+)
     private volatile boolean isClusterReady = false;
```

Comment:
Can we just remove `isClusterReady` altogether? We won't allow upgrading directly from 2.1 -> 4.1 right?